service/matching/physical_task_queue_manager.go

```diff
@@ -540,6 +527,79 @@ func (c *physicalTaskQueueManagerImpl) TrySyncMatch(ctx context.Context, task *i
 	return c.matcher.Offer(childCtx, task)
 }
 
+func (c *physicalTaskQueueManagerImpl) ensureRegisteredInDeployment(
+	ctx context.Context,
+	namespaceEntry *namespace.Namespace,
+	pollMetadata *pollMetadata,
+) error {
+	if !pollMetadata.workerVersionCapabilities.UseVersioning {
+		return nil
+	}
+	if !c.partitionMgr.engine.config.EnableDeployments(namespaceEntry.Name().String()) {
+		return nil
```

Comment:
if v3 is used but this config is not set, should we reject the pollers? Do we reject v1-2 pollers if versioning config is not enabled?

Reply:
we have the separate "workflow apis" setting... there's just one setting for v1 and v2 though.

I suppose the answer is yes? I'm not sure of all the implications though

Reply:
Still not sure why we have separate configs for v1-2. we always enable/disable them together.

Reply:
the idea was something like, if we find a bug, we can block further metadata updates without halting workflows